cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -110,6 +110,9 @@ type MasterConfiguration struct {
 
 	// The cluster name
 	ClusterName string
+
+	// SkipKubeProxyInstall indicates if the kube-proxy will be installed.
+	SkipKubeProxyInstall bool
```

Comment:
I'm not a fan of doing this, it would be better to cleanup phases to the point where folks can skip a step.

Reply:
@timothysc So, nesting the `SkipInstall` in the KubeProxy struct is better?

Reply:
@xlgao-zju No, you should be able explicitly run the phases you need and skip the proxy step.

Reply:
I am not quite sure about how to do this... Could you guys give me some guidance? @neolit123 @detiber @stealthybox

Reply:
@xlgao-zju phases are about to be moved and possibly refactored in 1.12. there could be a better way to do this instead of using the `KubeProxy.SkipInstall`  flag, so that's way Tim is putting a hold.

just wait on this PR and see if you can do something else from the backlog in the meantime. :)
take this if you want?
https://github.com/kubernetes/kubeadm/issues/935

Reply:
@neolit123 Thank you for your information. Is there any discussion of moving the phases? How long may I wait for this PR? Making kube-proxy optional is useful. And my team is looking forward to this feature. :P

>take this if you want?
kubernetes/kubeadm#935

Thank you for your suggestion. I'll take this. And feel free to ping me, if there is anything I can help. I'd love to help you guys. 😆

Reply:
@xlgao-zju 

> Thank you for your suggestion. I'll take this.

awesome.

> How long may I wait for this PR? Making kube-proxy optional is useful. 

i think i've seen the original phases proposal, but i don't have the link.
it's up to @luxas and @timothysc  to make the decision about this PR here. :)